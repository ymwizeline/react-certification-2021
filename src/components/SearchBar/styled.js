import styled from 'styled-components';

export const Form = styled.form`
  position: relative;
  width: 100%;
`;

export const Input = styled.input`
  margin-left: 30px;
  height: 100%;
  width: 100%;
  border-radius: 3px;
  background-color: ${(props) => props.theme.mainFade};
  color: ${(props) => props.theme.secondary};
  font-size: 1.2rem;
  border: none;
  outline: none;
  padding: 5px 20px 5px 45px;

  @media (max-width: 768px) {
    margin-left: 15px;
  }
`;

export const MagnifyingGlass = styled.div`
  position: absolute;
  font-size: 1.4rem;
  left: 40px;
`;
