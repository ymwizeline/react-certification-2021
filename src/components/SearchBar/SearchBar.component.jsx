import React from 'react';
import { Form, Input, MagnifyingGlass } from './styled';

const SearchBar = () => {
  return (
    <Form>
      <MagnifyingGlass>
        <span role="img" aria-label="magnifying-glass">
          🔎
        </span>
      </MagnifyingGlass>
      <Input />
    </Form>
  );
};

export default SearchBar;
