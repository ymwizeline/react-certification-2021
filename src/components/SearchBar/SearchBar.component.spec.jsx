import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from './SearchBar.component';
import 'jest-styled-components';

describe('SearchBar component', () => {
  it(`renders properly`, () => {
    const element = renderer.create(<SearchBar />).toJSON();

    expect(element).toMatchSnapshot();
  });
});
