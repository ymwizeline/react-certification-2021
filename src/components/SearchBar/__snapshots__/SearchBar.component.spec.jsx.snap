// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders properly 1`] = `
.c0 {
  position: relative;
  width: 100%;
}

.c2 {
  margin-left: 30px;
  height: 100%;
  width: 100%;
  border-radius: 3px;
  font-size: 1.2rem;
  border: none;
  outline: none;
  padding: 5px 20px 5px 45px;
}

.c1 {
  position: absolute;
  font-size: 1.4rem;
  left: 40px;
}

@media (max-width:768px) {
  .c2 {
    margin-left: 15px;
  }
}

<form
  className="c0"
>
  <div
    className="c1"
  >
    <span
      aria-label="magnifying-glass"
      role="img"
    >
      ðŸ”Ž
    </span>
  </div>
  <input
    className="c2"
  />
</form>
`;
